routers/web/admin/users.go

```diff
@@ -381,12 +381,8 @@ func EditUserPost(ctx *context.Context) {
 			return
 		}
 
-		if u.Salt, err = user_model.GetUserSalt(); err != nil {
-			ctx.ServerError("UpdateUser", err)
-			return
-		}
-		if err = u.SetPassword(form.Password); err != nil {
-			ctx.ServerError("SetPassword", err)
+		if err := user_service.ChangePassword(ctx, u, form.Password, u.MustChangePassword); err != nil {
```

Comment:
The behaviour has been changed. Before it will only change the memory value but not it will write to database. So it will be a bug from a user who changed password and other items of the user. I think the user interface need to be changed as well.

Reply:
Agh! Yes, lunny is right here. The previous code would only update the password on the db at line 455/451. 

Whilst it might make sense to only allow changing of password or changing of other things that's not quite what this code does. If there is a downstream error, say in getting the 2fa status then the password will still be changed but the user will not know.

This implies that there will need to be more UI changes.

Reply:
That should just be a transaction, then the order should be irrelavant.

Reply:
Could we don't update database here?

Reply:
@lunny Is it ok for you if that's not changed now because this place gets changed again in #28733? I will block the bot until you answer.